src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs

```diff
@@ -95,7 +95,11 @@ public static string GetAnalyzerAssemblyName(this DiagnosticAnalyzer analyzer)
 
         public static OptionSet GetAnalyzerOptionSet(this AnalyzerOptions analyzerOptions, SyntaxTree syntaxTree, CancellationToken cancellationToken)
         {
-            return GetAnalyzerOptionSetAsync(analyzerOptions, syntaxTree, cancellationToken).GetAwaiter().GetResult();
+            var optionSetAsync = GetAnalyzerOptionSetAsync(analyzerOptions, syntaxTree, cancellationToken);
+            if (optionSetAsync.IsCompleted)
+                return optionSetAsync.Result;
+
+            return optionSetAsync.AsTask().GetAwaiter().GetResult();
```

Comment:
@sharwell The change here is to avoid allocating a real Task if the ValueTask completed already?

Reply:
This is a correctness issue, not a performance issue. `ValueTask<TResult>.GetAwaiter().GetResult()` and `ValueTask<TResult>.Result` are only valid if the task is already complete. If you aren't sure whether it's complete, you have to use `AsTask()` to get a `Task<TResult>` and operate on that instead.